sdc11073/mdib/clientmdib.py

```diff
@@ -155,719 +163,754 @@ def get_age_stdev(self):
+                            state_container.descriptorHandle, state_container.DescriptorVersion,
+                            state_container.descriptor_container.DescriptorVersion)
+                        state_container.set_descriptor_container(None)
                     try:
-                        oldStateContainer = self.states.descriptorHandle.getOne(sc.descriptorHandle, allowNone=True)
-                    except RuntimeError  as ex:
-                        self._logger.error('_onEpisodicMetricReport, getOne on states: {}', ex)
+                        old_state_container = self.states.descriptorHandle.get_one(state_container.descriptorHandle,
+                                                                                   allow_none=True)
+                    except RuntimeError as ex:
+                        self._logger.error('_on_episodic_metric_report, get_one on states: {}', ex)
                         continue
-                    desc_h = sc.descriptorHandle
-                    metricsByHandle[desc_h] = sc  # metric
-                    if oldStateContainer is not None:
-                        if self._hasNewStateUsableStateVersion(oldStateContainer, sc, 'EpisodicMetricReport', is_buffered_report):
-                            oldStateContainer.updateFromOtherContainer(sc)
-                            self.states.updateObject(oldStateContainer)
+                    desc_h = state_container.descriptorHandle
+                    metrics_by_handle[desc_h] = state_container  # metric
+                    if old_state_container is not None:
+                        if self._has_new_state_usable_state_version(old_state_container, state_container,
+                                                                    'EpisodicMetricReport',
+                                                                    is_buffered_report):
+                            old_state_container.update_from_other_container(state_container)
+                            self.states.update_object(old_state_container)
                     else:
-                        self.states.addObject(sc)
+                        self.states.add_object(state_container)
 
-                    if sc.metricValue is not None:
-                        observationTime = sc.metricValue.DeterminationTime
-                        if observationTime is None:
+                    if state_container.MetricValue is not None:
+                        determination_time = state_container.MetricValue.DeterminationTime
+                        if determination_time is None:
                             self._logger.warn(
-                                '_onEpisodicMetricReport: metric {} version {} has no DeterminationTime',
-                                desc_h, sc.StateVersion)
+                                '_on_episodic_metric_report: metric {} version {} has no DeterminationTime',

```

Comment:
Added additional check of Validity and ActivationState. The place is ok, because this is only used to generate warnings if metrics are already too old when received.